src/components/Traceroute/Traceroute.tsx

```diff
@@ -55,89 +43,206 @@ const TracerouteTool = () => {
         switch (values.tracerouteSwitch) {
-            case "Traceroute ICMP scan": //traceroute syntax: traceroute -I <hostname>
+            case "Traceroute ICMP scan":
+                // Traceroute ICMP scan uses the '-I' option
+                // Syntax: traceroute -I <hostname>
                 args = [`/usr/share/ddt/Bash-Scripts/Tracerouteshell.sh`];
                 args.push(`-I`);
-                args.push(`${values.hostname}`);
+                args.push(`${values.hostname}`); // Adds the hostname to the arguments list.
                 try {
+                    // Executed the traceroute command using a shell script and captures the output.
                     let output = await CommandHelper.runCommand("bash", args);
-                    setOutput(output);
+                    setOutput(output); // Sets the output to the state.
+                    setAllowSave(true);
                 } catch (e: any) {
-                    setOutput(e);
+                    setOutput(e); // Sets error to the state if command fails.
                 }
 
                 break;
 
-            case "Traceroute TCP scan": //traceroute syntax: traceroute -T <hostname>
+            case "Traceroute TCP scan":
+                // TRaceroute TCP scan uses the '-T' option.
+                // Syntax: traceroute -T <hostname>
                 args = [`/usr/share/ddt/Bash-Scripts/Tracerouteshell.sh`];
                 args.push(`-T`);
-                args.push(`${values.hostname}`);
+                args.push(`${values.hostname}`); // Adds the hostname to the arguments list.
                 try {
                     let output = await CommandHelper.runCommand("bash", args);
-                    setOutput(output);
+                    setOutput(output); //Sets the output to the state.
+                    setAllowSave(true);
                 } catch (e: any) {
-                    setOutput(e);
+                    setOutput(e); //Sets error to the state if command fails.
                 }
 
                 break;
 
-            case "Traceroute UDP scan": //traceroute syntax: traceroute -U <hostname>
+            case "Traceroute UDP scan":
+                // Traceroute UDP scan uses the '-U' option.
+                // Syntax: traceroute -U <hostname>
                 args = [`/usr/share/ddt/Bash-Scripts/Tracerouteshell.sh`];
                 args.push(`-U`);
-                args.push(`${values.hostname}`);
+                args.push(`${values.hostname}`); // Adds the hostname to the arguments list.
                 try {
                     let output = await CommandHelper.runCommand("bash", args);
-                    setOutput(output);
+                    setOutput(output); // Sets the output to the state.
+                    setAllowSave(true);
                 } catch (e: any) {
-                    setOutput(e);
+                    setOutput(e); // Sets errors to the state if command fails.
                 }
 
                 break;
```

Comment:
Review requirement for a break this is likely not required in switch stack.

Reply:
I tested the break statement and I believe it is required as it negatively impacts the result and execution providing incorrect executions.